beetsplug/embedart.py

```diff
@@ -107,13 +111,43 @@ def embed_func(lib, opts, args):
                     art.embed_item(self._log, item, imagepath, maxwidth,
                                    None, compare_threshold, ifempty,
                                    quality=quality)
+            elif opts.url:
+                from PIL import Image
+                try:
+                    response = requests.get(opts.url, timeout=5)
+                    response.raise_for_status()
+                except requests.exceptions.RequestException as e:
+                    self._log.error("Error: {}".format(e))
+                try:
+                    img = Image.open(BytesIO(response.content))
+                except OSError as e:
+                    self._log.error("Error: {}".format(e))
+                if img.format:
```

Comment:
Also I don't know PIL well enough to know this but does the file extension matter? What if the url provides a png file. I'm assuming this check looks if it is "any supported image format", which is a good idea.

You could call the image something other then temp.jpg to make that clear.

Also talking about temp files. I think I saw some code in the beets playlist plugin where temp files are used and it uses a special module for that. Maybe look that up.

And since BytesIO gives us a temporary file already, just an idea and not sure if that's possible: Passing this object which behaves like a fileobject, as the docs say, to art.embed_item() directly doesn't work, does it? Guess it's expecting an actual file on disk right?

Reply:
Now, I manually specify the format, which will take care of this issue.